static/app/components/events/interfaces/spans/spanContext.tsx

```diff
@@ -0,0 +1,67 @@
+import {Component, createContext} from 'react';
+
+import {ProcessedSpanType} from './types';
+
+export type SpanContextProps = {
+  addExpandedSpan: (span: Readonly<ProcessedSpanType>, callback?: () => void) => void;
+  didAnchoredSpanMount: boolean;
+  isSpanExpanded: (span: Readonly<ProcessedSpanType>) => boolean;
+  markAnchoredSpanIsMounted: () => void;
+  removeExpandedSpan: (span: Readonly<ProcessedSpanType>, callback?: () => void) => void;
+};
+
+const SpanContext = createContext<SpanContextProps>({
```

Comment:
I'm using this context to manage two things:

- Keep track of all spans that are currently expanded, so we can expand them again in case they re-mount
- If a span is anchored, keep track of whether that span has mounted so we don't keep jumping to it whenever it re-mounts